.github/workflows/checkstyle.yaml

```diff
@@ -93,19 +96,21 @@ jobs:
           cat prettier-errors.txt
 
       - name: Check
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.BUILDBUDDY_GITHUB_USER_TOKEN }}
         run: |
           echo "===== gofmt diff (fix with ./buildfix.sh) ====="
-          cat gofmt-diff.txt
+          cat gofmt-diff.txt | reviewdog -reporter=github-pr-review -f=diff -f.diff.strip=1
           echo "===== buildifier diff (fix with ./buildfix.sh) ====="
-          cat buildifier-diff.txt
+          cat buildifier-diff.txt | reviewdog -reporter=github-pr-review -f=diff -f.diff.strip=1
           echo "===== gazelle diff (fix with ./buildfix.sh --gazelle) ====="
-          cat gazelle-diff.txt
+          cat gazelle-diff.txt | reviewdog -reporter=github-pr-review -f=diff -f.diff.strip=1
           echo "===== go deps diff (fix with ./buildfix.sh --go_deps) ====="
-          cat go-deps-diff.txt
+          cat go-deps-diff.txt | reviewdog -reporter=github-pr-review -f=diff -f.diff.strip=1
           echo "===== clang-format errors (fix with ./buildfix.sh) ====="
-          cat clang-format-errors.txt
+          cat clang-format-errors.txt | reviewdog -reporter=github-pr-review -f=diff -f.diff.strip=1
           echo "===== prettier errors (fix with ./buildfix.sh) ====="
-          cat prettier-errors.txt
+          cat prettier-errors.txt | reviewdog -reporter=github-pr-review -f=diff -f.diff.strip=1
```

Comment:
These 2 output errors, not a diff, so I am not sure what reviewdog will do in this case?

Reply:
Ah, It can annotate the line of code with such error as a normal comment (not diff suggestion).

But I should test if the error format is compatible with reviewdog first and adjust the flags accordingly 🤔 

Nice call!

Reply:
I added a quick example for clang error format. Turn out it's compatible with staticcheck.

I disabled prettier. There is this example on how to use prettier with reviewdog but I am a bit too dull to understand it https://github.com/EPMatt/reviewdog-action-prettier/blob/main/script.sh 🤔 

Perhaps we could include it in a future PR.